### PR TITLE
Make regression testing faster.

### PR DESCRIPTION
**Pull request by paul-guo-**

### CODE REVIEW

File: src/test/regress/greenplum_schedule [hunk 1]
```diff
@@ -14,6 +14,13 @@
 #   to the segments, and therefore has to run in smaller groups to avoid
 #   hitting max_connections limit on segments.
 #
+
+# gp_toolkit performs a vacuum and checks that it truncated the relation. That
+# might not happen if other backends are holding transactions open, preventing
+# vacuum from removing dead tuples. And run gp_toolkit early to make some log
+# file related tests (like gp_log_system, etc) faster.
```
Comment (danielgustafsson) [inline src/test/regress/greenplum_schedule]: The `gp_log_database` test queries are quite uninteresting as they are today. Would it make them faster to rewrite them to something like `select count(*) > 0 from gp_toolkit.gp_log_database where logseverity='LOG';`?

Comment (paul-guo-) [reply src/test/regress/greenplum_schedule]: Tested, no. And ·limit· is faster （see plans below) although in theory count(*) could be faster.
 
regression=# select logseverity from gp_toolkit.gp_log_database where logseverity='LOG' limit 3;
 logseverity
-------------
 LOG
 LOG
 LOG
(3 rows)

Time: 575.896 ms
regression=# select count(*) from gp_toolkit.gp_log_database where logseverity='LOG';
 count
-------
 12843
(1 row)

Time: 594.644 ms
regression=# explain select count(*) from gp_toolkit.gp_log_database where logseverity='LOG';
                                               QUERY PLAN
--------------------------------------------------------------------------------------------------------
 Aggregate  (cost=16002.27..16002.28 rows=1 width=8)
   ->  Sort  (cost=16001.78..16001.86 rows=33 width=800)
         Sort Key: __gp_log_segment_ext.logtime
         ->  Append  (cost=0.00..16000.96 rows=33 width=800)
               ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..16000.95 rows=32 width=800)
                     ->  External Scan on __gp_log_segment_ext  (cost=0.00..16000.00 rows=11 width=800)
                           Filter: ((logdatabase = 'regression'::text) AND (logseverity = 'LOG'::text))
               ->  External Scan on __gp_log_master_ext  (cost=0.00..0.00 rows=1 width=800)
                     Filter: ((logdatabase = 'regression'::text) AND (logseverity = 'LOG'::text))
 Optimizer: legacy query optimizer
(10 rows)

Time: 18.947 ms
regression=# explain select logseverity from gp_toolkit.gp_log_database where logseverity='LOG' limit 3;
                                                  QUERY PLAN
--------------------------------------------------------------------------------------------------------------
 Limit  (cost=16001.78..16001.82 rows=3 width=32)
   ->  Subquery Scan on gp_log_system  (cost=16001.78..16002.19 rows=33 width=32)
         ->  Sort  (cost=16001.78..16001.86 rows=33 width=800)
               Sort Key: __gp_log_segment_ext.logtime
               ->  Append  (cost=0.00..16000.96 rows=33 width=800)
                     ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..16000.95 rows=32 width=800)
                           ->  External Scan on __gp_log_segment_ext  (cost=0.00..16000.00 rows=11 width=800)
                                 Filter: ((logdatabase = 'regression'::text) AND (logseverity = 'LOG'::text))
                     ->  External Scan on __gp_log_master_ext  (cost=0.00..0.00 rows=1 width=800)
                           Filter: ((logdatabase = 'regression'::text) AND (logseverity = 'LOG'::text))
 Optimizer: legacy query optimizer
(11 rows)

Time: 21.407 ms

**Comment by hlinnaka** [comment]
Cool, faster is better!

I wonder how useful those log-scanning functions are in production, if we're already struggling with them in the regression tests. Or perhaps the regression tests create much more logs than a production system would. In any case, +1 for this patch, there's no reason to not make regression tests go fast if we can.

**Comment by paul-guo-** [comment]
Thanks for the review. Pipeline fails in gpexpand only but that has nothing to do with regression tests. So I'll check in immediately.

**Pull request merged**